### DISCUSSION TITLE
Slow Batch Requests

### DISCUSSION THREAD

**Discussion by suelitonx** [pocketbase/pocketbase#6035] [Q&A]
Hello, I understand that the number of columns can impact the performance/speed of batch requests, but is it normal for processing 50,000 records with 9 columns to take 10 minutes?

I’ve already adjusted the settings to allow for larger batches and longer processing times, but it’s still quite slow. Since I’m running it locally, I expected the process to be faster.

For context, I’m using a machine with 24 GB of RAM and a Ryzen 5 1600 processor, running on Windows.

Does anyone have tips on how to improve performance?

**Comment by ganigeorgiev** [comment]
Without details how to replicate it I don't think anyone will be able to help you and I hate performance discussions when there is nothing concrete to discuss.

Feel free to provide more details how you are sending the request, what is the data and the overall total body size you are submitting so that me or someone else can replicate it.

In general I'll advice against using too much large allowance for the batch limits because as mentioned in the documentation and the API Preview it could degrade the performance of your application.

**Reply by ganigeorgiev** [reply]
Note also that the batch Web API does a lot of copying and in addition it will trigger all associated hooks in your case 50k times (each create request will fire ~10 hooks * 50k) and this could be slow. Even returning the serialized json of the combined 50k records response could take more than a second depending on the data.

Again, free to provide more details and elaborate what you are trying to do but submitting so much data is not a use case for the batch Web API and you may have to create your own custom route that will process the record in a single transaction without firing all other hooks and maybe returning 204 with empty body instead of the inserted data.
